code_gen/code_gen/generate_component_code.py

```diff
@@ -0,0 +1,43 @@
+import helpers.langchain_helpers as langchain_helpers
+import helpers.supabase_helpers as supabase_helpers
+from dotenv import load_dotenv
+load_dotenv()
+
+
+def main(app, prompt, verbose=False):
+    use_docs = True
+    db = supabase_helpers.SupabaseConnector()
+    row = db.client.table('components').select('*').match({'app_name_slug': app}).execute().data[0]
+
+    if row is None:
+        return langchain_helpers.no_docs(app, prompt, verbose)
+
+    if row['docs_url']:
```

Comment:
I'd recommend refactoring the logic within the `docs_url` and `openapi_url` into their own functions too